File: genetic-packer/src/main/java/genetic/packer/dto/response/TranslatedBoxDto.java
package genetic.packer.dto.response;

import genetic.api.builder.HasBuilder;
import genetic.packer.dto.BoxDto;

/**
 * @author piotr.larysz
 */
@HasBuilder
public class TranslatedBoxDto extends BoxDto {

    private TranslationDto<Integer> position;

    private TranslationDto<Double> rotation;

    public TranslationDto<Integer> getPosition() {
        return position;
    }

    public void setPosition(TranslationDto<Integer> position) {
        this.position = position;
    }

    public TranslationDto<Double> getRotation() {
        return rotation;
    }

    public void setRotation(TranslationDto<Double> rotation) {
        this.rotation = rotation;
    }
}

File: genetic-packer/src/main/java/genetic/packer/dto/response/TranslatedBoxDto.kt
package genetic.packer.dto.response

import genetic.packer.dto.BoxDto

data class TranslatedBoxDto(
    override val width: Int,
    override val height: Int,
    override val depth: Int,
    val position: TranslationDto<Int>,
    val rotation: TranslationDto<Double>? = null
) : BoxDto